### PR TITLE
Add build script

### PR DESCRIPTION
**Pull request by TangRufus**
#### Usage 1

```bash
composer install
./bin/build <git-remote>
```

Example:
```bash
composer install
./bin/build git@github.com:TangRufus/testing-roots-wordpress-packager.git
./bin/build https://github.com/TangRufus/testing-roots-wordpress-packager.git
./bin/build /path/to/local/git/repo
```

#### Usage 2

```bash
composer run-script --timeout=<seconds> build <git-remote>
```

Example:
```bash
composer run-script --timeout=1800 build git@github.com:TangRufus/testing-roots-wordpress-packager.git
composer run-script --timeout=1800 build https://github.com/TangRufus/testing-roots-wordpress-packager.git
composer run-script --timeout=1800 build /path/to/local/git/repo
```

#### Result

https://github.com/TangRufus/testing-roots-wordpress-packager

---

#### CircleCI

Config: https://github.com/TangRufus/testing-roots-wordpress-packager/blob/build/.circleci/config.yml

Result: https://circleci.com/gh/TangRufus/testing-roots-wordpress-packager/56

---

#### Help Wanted

- [ ] Allow docker image to read SSH keys (or HTTPS credentials)
- [ ] Run build command inside docker image
- [ ] Add build command to Makefile
- [x] Add license
- [ ] Test `\Roots\WordPressPackager\Build`
- [x] Test `\Roots\WordPressPackager\Target`

@austinpray Can you help on these?

### CODE REVIEW

**Comment by austinpray** [comment]
Yes I can help!

**Comment by austinpray** [comment]
Throwing some PRs over at https://github.com/TangRufus/roots-wordpress-packager/pulls

File: src/WordPressPackager/Target.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            $composerJsonContent
+        );
+
+        $this->gitRepo->addFile('composer.json');
+        $this->gitRepo->commit("Version bump ${version}");
+        $this->gitRepo->createTag($version, [
+            '--annotate',
+            '--message' => "Version bump ${version}",
+        ]);
+        $this->gitRepo->push('origin', ["refs/heads/${version}", '--follow-tags']);
```
Comment (LeoColomb) [inline src/WordPressPackager/Target.php]: Maybe a dumb question, but why do you push tags AND branches when composer only use tags for releases?

```suggestion
        $this->gitRepo->push('origin', ["refs/tags/${version}"]);
```

Comment (TangRufus) [reply src/WordPressPackager/Target.php]: Good catch! Pushing branches is unnecessary

**Comment by TangRufus** [comment]
New changes pushed and squash:

- Echo MIT License
- Test \Roots\WordPressPackager\Target
- Push tags only (without branches) (thanks @LeoColomb )

---

These tasks need help:

- Allow docker image to read SSH keys (or HTTPS credentials)
- Run build command inside docker image
- Add build command to Makefile
- Test \Roots\WordPressPackager\Build (e2e tests)

Question: Should we merge this pull request and leave these tasks to developers smarter than me?

@austinpray 

---

Probably needed in the future:

- Accept different sources (wp.org HTML, https://github.com/wordpress/wordpress, etc) as build command arguments
- Accept composer package name as build command arguments
- Accept license author name as build command arguments

**Comment by austinpray** [comment]
@tangrufus want to merge and then create the remaining todos as issues? I can take care of those.

**Pull request merged**